### PR TITLE
Refactor the toolkit to use the new Python_Toolkit

### LINKED ISSUE #6: Use Python_Toolkit instead of the Python.Included

### PR DESCRIPTION
**Pull request by epignatelli**
### NOTE: Depends on 
https://github.com/BHoM/Python_Toolkit/pull/2
Despite the code itself does not have dependency, this pr is coupled with https://github.com/BHoM/Numpy_Toolkit/pull/9.
It will compile if you're not on that branch, but the toolkit will not work.

   
### Issues addressed by this PR

Closes #6

### Test files

This requires a bit of a clean up to avoid conflicts. 
1. Remove the `BHoM/Assemblies` folder
2. Compile the Python_Toolkit
3. Compile the Numpy_Toolkit
4. Compile the Keras_Toolkit
4. Use any test script from the Numpy of Keras library, e.g. https://burohappold.sharepoint.com/:f:/s/BHoM/Ei4hzzyRJqNPpqF_CHW4EmIBLrFsCtg33CEeYRZdTIvh4A?e=j7Gljx

### CODE REVIEW

**Comment by epignatelli** [comment]
CTRL C + CTRL V
@alelom @FraserGreenroyd all ready for review! Let me know if you need help with that!

File: Keras/Keras.csproj [hunk 1]
```diff
@@ -45,15 +45,19 @@
...
-    <PackageReference Include="BH.UI.Python.Runtime" Version="3.7.4-alpha" />
     <PackageReference Include="Microsoft.CSharp" Version="4.6.0" />
   </ItemGroup>
 
   <ItemGroup>
     <Reference Include="Numpy">
       <HintPath>..\..\Numpy_Toolkit\Build\Numpy.dll</HintPath>
     </Reference>
+    <Reference Include="Python.Runtime">
+      <HintPath>..\..\Python_Toolkit\Build\Python.Runtime.dll</HintPath>
```
Comment (FraserGreenroyd) [inline Keras/Keras.csproj:56]: Copy local should be set to false

File: Keras/Keras.csproj [hunk 2]
```diff
@@ -45,15 +45,19 @@
...
 
   <ItemGroup>
-    <PackageReference Include="BH.UI.Python.Included" Version="3.7.3.9-alpha" />
-    <PackageReference Include="BH.UI.Python.Runtime" Version="3.7.4-alpha" />
     <PackageReference Include="Microsoft.CSharp" Version="4.6.0" />
   </ItemGroup>
 
   <ItemGroup>
     <Reference Include="Numpy">
       <HintPath>..\..\Numpy_Toolkit\Build\Numpy.dll</HintPath>
```
Comment (FraserGreenroyd) [inline Keras/Keras.csproj:53]: Copy local should be set to false

File: Keras/Keras.csproj [hunk 3]
```diff
@@ -45,15 +45,19 @@
...
     </Reference>
+    <Reference Include="Python.Runtime">
+      <HintPath>..\..\Python_Toolkit\Build\Python.Runtime.dll</HintPath>
+    </Reference>
+    <Reference Include="Python_Engine">
+      <HintPath>..\..\Python_Toolkit\Build\Python_Engine.dll</HintPath>
```
Comment (FraserGreenroyd) [inline Keras/Keras.csproj:59]: Copy local should be set to false

File: Keras_Engine/Keras_Engine.csproj [hunk 4]
```diff
@@ -76,11 +76,16 @@
...
-    <Reference Include="Python.Included, Version=3.7.9.2, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\BH.UI.Python.Included.3.7.3.9-alpha\lib\netstandard2.0\Python.Included.dll</HintPath>
+    <Reference Include="Numpy, Version=3.7.1.16, Culture=neutral, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\Numpy_Toolkit\Build\Numpy.dll</HintPath>
     </Reference>
     <Reference Include="Python.Runtime, Version=3.7.4.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\BH.UI.Python.Runtime.3.7.4-alpha\lib\netstandard2.0\Python.Runtime.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\Python_Toolkit\Build\Python.Runtime.dll</HintPath>
```
Comment (FraserGreenroyd) [inline Keras_Engine/Keras_Engine.csproj:86]: Copy local should be set to false

File: Keras_Engine/Keras_Engine.csproj [hunk 5]
```diff
@@ -76,11 +76,16 @@
...
+    </Reference>
+    <Reference Include="Python_Engine">
+      <HintPath>..\..\Python_Toolkit\Build\Python_Engine.dll</HintPath>
```
Comment (FraserGreenroyd) [inline Keras_Engine/Keras_Engine.csproj:90]: Copy local should be set to false

File: Keras_Engine/Keras_Engine.csproj [hunk 6]
```diff
@@ -116,4 +122,8 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <PropertyGroup>
+    <PostBuildEvent>rem Installing keras in python at compile time
+$(SolutionDir)/../Python_Toolkit/Python_PostBuild/bin/Debug/Python_Postbuild.exe "pillow" "tensorflow==2.0"</PostBuildEvent>
```
Comment (FraserGreenroyd) [inline Keras_Engine/Keras_Engine.csproj]: This wasn't working for me - but if I removed `Installing keras in python at compile time` then it did. With that in, it had a 9009 error, but removing that would make it build (but Python did not install - more on that in Python_Toolkit).

File: Keras_Engine/Query/Load.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Keras_Engine/Query/Load.cs]: Copyright now needs to be to 2020

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
I could not get this to work in the way described in the PR body, I could not get Python installer using the dependent toolkits, nor test Keras_Engine itself I'm afraid. See some comments on the code - might be worth going through this together at some point @epignatelli ?

**Review by FraserGreenroyd — APPROVED** [review]
Have re run the steps in the PR body and am happy this is now working following the changes @epignatelli

**Comment by epignatelli** [comment]
Thanks @FraserGreenroyd!

@alelom since this will be pretty foundational, would you mind trying this as well?

We should really have something that reflects the comments when solving PRs that span across different repos, but are actually the same 😄

**Comment by FraserGreenroyd** [comment]
> We should really have something that reflects the comments when solving PRs that span across different repos, but are actually the same 😄

🤣 agreed! I almost wore out my Ctrl C + V buttons on these PRs!

**Comment by epignatelli** [comment]
We might have the bot listening to the comments that happen on prs with a specific tag, like
`Linked to: X/pulls/Y`?

**Comment by epignatelli** [comment]
https://github.com/BHoM/Numpy_Toolkit/pull/12#issuecomment-586148394

**Comment by alelom** [comment]
Adding to approvals as requested image

**Comment by epignatelli** [comment]
Lovely, thanks @alelom !!

**Pull request merged**